app/cdap/components/DirectiveInput/index.tsx

```diff
@@ -0,0 +1,263 @@
+        payload: true,
+      });
+      dispatch({
+        type: IDirectiveActions.APPLIED_DIRECTIVE,
+        payload: [inputText[0]],
+      });
+    }
+    if (isDirectiveSet) {
+      columnNamesList.forEach((column) => {
+        if (value.indexOf(column.label) !== -1) {
+          setDirectiveColumns((prev) =>
+            !prev.includes(column.label) ? [...prev, column.label] : prev
+          );
+        } else {
+          setDirectiveColumns((prev) => prev.filter((el) => el != column.label));
+          setEnterCount(appliedDirective.length + directiveColumns.length);
+        }
+      });
+    }
+    dispatch({
+      type: IDirectiveActions.INPUT_DIRECTIVE,
+      payload: value,
+    });
+  };
+
+  useEffect(() => {
+    dispatch({
+      type: IDirectiveActions.DIRECTIVE_USAGE_LIST,
+      payload: formatDirectiveUsageData(inputDirective, directivesList),
+    });
+  }, [inputDirective]);
+
+  const handleKeyDownEvent = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.keyCode === 86) {
+      dispatch({
+        type: IDirectiveActions.DIRECTIVE_PASTE,
+        payload: true,
+      });
+    } else if (event.key === 'Enter') {
+      setEnterCount((prev) => {
+        if (
+          isDirectiveSet &&
+          directiveColumns.length >= 1 &&
+          directiveColumnCount === 0 &&
+          prev + 1 > directiveColumns.length + appliedDirective.length
+        ) {
+          // This condition means if we can select multiple column for directive
+          onDirectiveInputHandler(inputDirective);
+        } else if (
+          isDirectiveSet &&
+          directiveColumns.length === 2 &&
+          directiveColumnCount === 2 &&
+          prev + 1 > directiveColumns.length + appliedDirective.length
+        ) {
+          // This condition means if we can select atleast two column for directive
+          onDirectiveInputHandler(inputDirective);
+        } else if (prev + 1 > directiveColumns.length + appliedDirective.length) {
+          // This condition means if we can select single column from list and any number of postfix can be entered
+          onDirectiveInputHandler(inputDirective);
+        } else if (isDirectivePaste) {
+          // If we are copy pasting directive this condition is executed
+          onDirectiveInputHandler(inputDirective);
+        }
+        return prev + 1;
+      });
+    }
+  };
+
+  return (
+    <DirectiveBox data-testid="directive-input-main-container">
+      <InputParentWrapper data-testid="directive-input-parent">
+        <InputPanel
+          inputDirective={inputDirective}
+          onSearchItemClick={(value) => handleDirectiveChange(value)}
+          setDirectivesList={(values) =>
+            dispatch({
+              type: IDirectiveActions.DIRECTIVE_LIST,
+              payload: values,
+            })
+          }
+          getDirectiveSyntax={(activeResults: IDirectiveUsage[], isDirectiveSelected) => {
+            dispatch({
```

Comment:
again here these two dispatch calls can be combined

Reply:
Done